src/Common/Crypto/CryptoGen.php

```diff
@@ -208,13 +226,13 @@ private function coreEncrypt($sValue, $customPassword = null, $keySource = 'driv
      * Function to decrypt data
      * Should not be called directly (only called by decryptStandard() function)
      *
-     * @param  string  $sValue          Encrypted data that will be decrypted.
-     * @param  string  $customPassword  If null, then use standard keys. If provide a password, then will derive key from this.
-     * @param  string  $keySource       This is the source of the keys. Options are 'drive' and 'database'
-     * @param  string  $keyNumber       This is the key number/version.
-     * @return string or false          returns the decrypted data or false if failed.
+     * @param string      $sValue         Encrypted data that will be decrypted.
+     * @param string|null $customPassword If null, then use standard keys. If provide a password, then will derive key from this.
+     * @param string      $keySource      This is the source of the keys. Options are 'drive' and 'database'
+     * @param string|null $keyNumber      This is the key number/version.
+     * @return false|string or false      returns the decrypted data or false if failed.
      */
-    private function coreDecrypt($sValue, $customPassword = null, $keySource = 'drive', $keyNumber = null)
+    private function coreDecrypt(string $sValue, string $customPassword = null, string $keySource = 'drive', string $keyNumber = null): false|string
```

Comment:
change $customPassword and $keyNumber to `?string`
may make sense to change $value to that also since guessing null values may get in there sometimes which would return empty string i think